test/unit-test/ota_utest.c

```diff
@@ -966,10 +1042,56 @@ void test_OTA_ReceiveFileBlockComplete()
     TEST_ASSERT_EQUAL( OtaAgentStateWaitingForJob, OTA_GetState() );
 }
 
+void test_OTA_ReceiveFileBlockCompleteHttp()
+{
+    OtaEventMsg_t otaEvent = { NULL, OtaAgentEventReceivedFileBlock };
+    uint8_t pFileBlock[ OTA_FILE_BLOCK_SIZE ] = { 0 };
+    int remainingBlocks = OTA_TEST_FILE_SIZE;
+    int idx = 0;
+
+    pOtaJobDoc = JOB_DOC_HTTP;
+    otaGoToState( OtaAgentStateWaitingForFileBlock );
+    TEST_ASSERT_EQUAL( OtaAgentStateWaitingForFileBlock, OTA_GetState() );
+
+    /* Set the event send interface to a mock function that allows events to be sent continuously
+     * because we're receiving multiple blocks in this test. */
+    otaInterfaces.os.event.send = mockOSEventSend;
+
+    /* Fill the file block. */
+    for( idx = 0; idx < sizeof( pFileBlock ); idx++ )
+    {
+        pFileBlock[ idx ] = idx % UINT8_MAX;
+    }
+
+    while( remainingBlocks > OTA_FILE_BLOCK_SIZE )
```

Comment:
Can you rename the remainingBlocks variable? This is a confusing name. It is not the number of blocks that are remaining, it is the number of bytes remaining (assuming that OTA_FILE_BLOCK_SIZE is in terms of bytes). Alternatively this could be changed to actually represent the number of remaining blocks by initializing it to the file size mod the block size, then changing how it's decremented.

Reply:
will use `remainingBytes` instead.